sql/src/main/java/io/crate/execution/dml/upsert/InsertSourceFromCells.java

```diff
@@ -118,6 +121,8 @@ public BytesReference generateSource(Object[] values) throws IOException {
             ColumnIdent column = entry.getKey().column();
             Maps.mergeInto(source, column.name(), column.path(), entry.getValue().value());
         }
+        // validate a source with injected generated column values
```

Comment:
```suggestion
```

Comment repeats the code and doesn't explain why. I think instead of adding a comment here we could add a docstring to `.validate`  that elaborates a bit that for the constraint check to work for generated columns, the values for them have to be injected into the provided source.